src/mgmt/cli_query.c

```diff
@@ -143,3 +143,118 @@ dc_mgmt_get_bs_state(tse_task_t *task)
 	tse_task_complete(task, rc);
 	return rc;
 }
+
+struct mgmt_get_vos_state_arg {
+	struct dc_mgmt_sys	*sys;
+	crt_rpc_t		*rpc;
+	int			*vos_state;
+	d_rank_t		 rank;
+	d_rank_t		 tgt;
+};
+
+static int
+mgmt_get_vos_state_cp(tse_task_t *task, void *data)
+{
+	struct mgmt_get_vos_state_arg	*arg;
+	struct mgmt_get_vos_state_out	*vs_out;
+	struct mgmt_get_vos_state_in	*vs_in;
+	int				 rc = task->dt_result;
+
+	arg = (struct mgmt_get_vos_state_arg *)data;
+
+	if (rc) {
+		D_ERROR("RPC error while querying VOS state: "DF_RC"\n",
+			DP_RC(rc));
+		D_GOTO(out, rc);
+	}
+
+	vs_out = crt_reply_get(arg->rpc);
+	D_ASSERT(vs_out != NULL);
+	rc = vs_out->vs_rc;
+	if (rc) {
+		D_ERROR("MGMT_GET_VOS_STATE replied failed, rc: %d\n", rc);
+		D_GOTO(out, rc);
+	}
+
+	*arg->vos_state = vs_out->vs_state;
+
+	vs_in = crt_req_get(arg->rpc);
+	D_ASSERT(vs_in != NULL);
+
+out:
+	dc_mgmt_sys_detach(arg->sys);
+	crt_req_decref(arg->rpc);
+
+	return rc;
+}
+
+int
+dc_pool_get_vos_state(tse_task_t *task)
```

Comment:
Let's move this implementation to dc_pool_query_target() in src/pool/cli.c, and get pool target state only for this moment. Let's improve it later to add per target space usage information.